medusa/process_tv.py

```diff
@@ -16,670 +15,596 @@
+        :param delete_on: Boolean for whether or not it should delete files
+        :param failed: Boolean for whether or not the download failed
+        :param proc_type: Type of postprocessing auto or manual
+        :param ignore_subs: True to ignore setting 'postpone if no subs'
+        """
+        if not self.directory:
+            return self.output
+
+        if app.POSTPONE_IF_NO_SUBS:
+            self._log("Feature 'postpone post-processing if no subtitle available' is enabled.")
+
+        for path in self.paths:
+
+            if not self.should_process(path, nzb_name, failed):
+                continue
 
-        try:
-            logger.log(u"Deleting folder (if it's empty): %s" % folder)
-            os.rmdir(folder)
-        except (OSError, IOError) as e:
-            logger.log(u"Warning: unable to delete folder: %s: %s" % (folder, ex(e)), logger.WARNING)
-            return False
-    else:
-        try:
-            logger.log(u"Deleting folder: " + folder)
-            shutil.rmtree(folder)
-        except (OSError, IOError) as e:
-            logger.log(u"Warning: unable to delete folder: %s: %s" % (folder, ex(e)), logger.WARNING)
-            return False
+            self.result = True
 
-    return True
+            for dir_path, filelist in self._get_files(path):
 
+                sync_files = (filename
+                              for filename in filelist
+                              if is_sync_file(filename))
+                # Don't process files if they are still being synced
+                postpone = app.POSTPONE_IF_SYNC_FILES and any(sync_files)
 
-def delete_files(processPath, notwantedFiles, result, force=False):
-    """
-    Remove files from filesystem.
+                if not postpone:
 
-    :param processPath: path to process
-    :param notwantedFiles: files we do not want
-    :param result: Processor results
-    :param force: Boolean, force deletion, defaults to false
-    """
-    if not result.result and force:
-        result.output += logHelper(u"Forcing deletion of files, even though last result was not successful", logger.DEBUG)
-    elif not result.result:
-        return
+                    self._log('\nProcessing folder: {0}'.format(dir_path), logger.DEBUG)
 
-    # Delete all file not needed
-    for cur_file in notwantedFiles:
+                    self.prepare_files(dir_path, filelist, force)
 
-        cur_file_path = os.path.join(processPath, cur_file)
+                    self.process_files(dir_path, nzb_name=nzb_name, force=force, is_priority=is_priority,
+                                       ignore_subs=ignore_subs)
 
-        if not os.path.isfile(cur_file_path):
-            continue  # Prevent error when a notwantedfiles is an associated files
+                    # Always delete files if they are being moved or if it's explicitly wanted
+                    if not self.process_method == 'move' or (proc_type == 'manual' and not delete_on):
```

Comment:
so need to change the label in manual PP page because its not clear

```(Check this to delete files and folders like auto processing)```

auto processing DOESN'T delete files with delete on and it never deleted my files